src/yunohost/app.py

```diff
@@ -2525,128 +2736,14 @@ def _parse_args_in_yunohost_format(user_answers, argument_questions):
                               format from actions.json/toml, manifest.json/toml
                               or config_panel.json/toml
     """
-    from yunohost.domain import domain_list, _get_maindomain
-    from yunohost.user import user_list, user_info
-
     parsed_answers_dict = OrderedDict()
 
     for question in argument_questions:
-        question_name = question['name']
-        question_type = question.get('type', 'string')
-        question_default = question.get('default', None)
-        question_choices = question.get('choices', [])
-        question_value = None
-
-        # Transpose default value for boolean type and set it to
-        # false if not defined.
-        if question_type == 'boolean':
-            question_default = 1 if question_default else 0
-
-        # do not print for webadmin
-        if question_type == 'display_text' and msettings.get('interface') != 'api':
-            print(_value_for_locale(question['ask']))
-            continue
+        parser = ARGUMENTS_TYPE_PARSERS[question.get("type", "string")]()
 
-        # Attempt to retrieve argument value
-        if question_name in user_answers:
-            question_value = user_answers[question_name]
-        else:
-            if 'ask' in question:
-
-                if question_type == 'domain':
-                    question_default = _get_maindomain()
-                    msignals.display(m18n.n('domains_available'))
-                    for domain in domain_list()['domains']:
-                        msignals.display("- {}".format(domain))
-
-                elif question_type == 'user':
-                    msignals.display(m18n.n('users_available'))
-                    users = user_list()['users']
-                    for user in users.keys():
-                        msignals.display("- {}".format(user))
-
-                    root_mail = "root@%s" % _get_maindomain()
-                    for user in users.keys():
-                        if root_mail in user_info(user)["mail-aliases"]:
-                            question_default = user
-                            break
```

Comment:
That stuff about default answer for user-type question disappeared during conflict/merge, looking at the new structure it should be pretty straightforward to reintegrate but busy right now with other reviews

Reply:
Ah indeed, quite a lot of other PR has modified this code, sorry for not moving fast enough :< 

(don't hesitate to push on it ™)